charts/pega/templates/_pega-deployment.tpl

```diff
@@ -163,6 +163,36 @@ spec:
         - name: REQUESTOR_PASSIVATION_TIMEOUT
           value: "{{ .node.requestor.passivationTimeSec }}"
 {{- end }}
+{{- if .node.dds }}
```

Comment:
These settings are already added in `pega-environment-config.yaml` config map. Is is required to define them the deployment envs again? Per [this](https://github.com/pegasystems/pega-helm-charts/pull/497/files#diff-d20d9d381fee827b6a82e73797e5f07a86abfb7a208931ac0164a558178a90d3L186) all the entries in config maps are loaded to deployments envs.

Reply:
This is intentional. Our requirement is to be able to specify these new configuration parameters system wide, but also on a per tier basis. I.e. at the same level as the node type. For example, we may want to enable speculative execution on decision hub nodes to ensure low latencies to Cassandra, but leave it disabled on data flow nodes to avoid overloading the Cassandra cluster during data flow runs.

Reply:
I have two questions.
1. In the below snippet, params are referred from `.node.dds`. To this context .node is tier object. Is it that, to override at tier level these params are supposed to define in `.Values.tier[].dds.`? If my understanding is right, Why is there no mention of this configuration in 'Values.yaml`, do you want to keep them as a hidden feature?
2. Deployment object going to contain tier level overidden cassandra params along with global cassandra params coming from config map. There will be two entries for each param in the evn list. Have you verified which one is taking precedence?

Reply:
The new configuration parameters are included for completeness. They are very rarely used in practice and would only be recommended based on detailed performance analysis of a system. Our desire is not to keep them hidden, but rather not promote their use where not required. The README.md file includes a reference example of how they may be specified at the tier level.

We have verified that the tier level settings take precedence over the global settings as per our requirement.

Reply:
Thanks for the confirmation. This actually, brings in a use case of the tier-level override of environment settings. This feature is not yet available, I would prefer to provide it as generic feature rather limited to DDS. May be like `.Values.tier[].env,`
@dcasavant @petejo @pega-chikv  What are your views on this?

Reply:
@kishorv10 Is it possible for you to approve this PR and if you wish to provide tier-level overrides as a generic feature we can move the DDS settings to be inline with any new approach at that time? Many thanks, Paul

Reply:
Created enhancement request [Issue-499](https://github.com/pegasystems/pega-helm-charts/issues/499) to have Pega helm charts support tier-level environment settings override. This PR will be on hold until the enhancement is prioritized and made available.

Reply:
@baylp2022  We realized that there is already feature (not exposed for general use) to pass custom environment variables at tier level, 
```
      livenessProbe:
        port: 8081

      custom:
        env: 
        - name: CENV_KEY1
          value: CValue1
        - name: CENV_KEY2
          value: CValue2
```
Can you please check if this helps your usecase?